pkg/kubectl/cmd/util/factory.go

```diff
@@ -214,13 +216,25 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			}
 			return kubectl.ReaperFor(mapping.Kind, client)
 		},
-		Validator: func(validate bool) (validation.Schema, error) {
+		Validator: func(validate bool, cacheDir string) (validation.Schema, error) {
 			if validate {
 				client, err := clients.ClientForVersion("")
 				if err != nil {
 					return nil, err
 				}
-				return &clientSwaggerSchema{client, client.ExperimentalClient, api.Scheme}, nil
+				version, err := client.ServerVersion()
```

Comment:
in general, I'm trying to eliminate the overhead involved in every invocation... adding a server version check to avoid a server api schema pull is a net win, I guess, but feels like two steps forward and one step back